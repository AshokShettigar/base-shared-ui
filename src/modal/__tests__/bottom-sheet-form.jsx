import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BUTTON_VARIATION } from 'platform-common-ui';
import BottomSheetForm, { Form } from 'platform-common-ui/modal/bottom-sheet-form';

const mockStore = configureStore();
describe('modal/bottom-sheet-form', function() {
    const ContentComponent = () => <div>Content</div>;
    const Component = Form(ContentComponent);

    it('should render the component', function() {
        const handler = () => true;
        const controls = [{
            label: 'Save',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }, {
            label: 'Cancel',
            onClick: handler,
            variation: BUTTON_VARIATION.SECONDARY
        }];

        const wrapper = shallow(<Component formControls={controls} sidebar="Sidebar Content" />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component without sidebar', function() {
        const handler = () => true;
        const controls = [{
            label: 'Save',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }, {
            label: 'Cancel',
            onClick: handler,
            variation: BUTTON_VARIATION.SECONDARY
        }];

        const wrapper = shallow(<Component formControls={controls} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with sub control', function() {
        const handler = () => true;
        const controls = [{
            label: 'Save',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }, {
            label: 'Cancel',
            onClick: handler,
            variation: BUTTON_VARIATION.SECONDARY
        }];

        const wrapper = shallow(
            <Component
                formControls={controls}
                subControl="Some control"
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the HOC component', function() {
        const handler = () => true;
        const controls = [{
            label: 'Save',
            onClick: handler,
            variation: BUTTON_VARIATION.PRIMARY
        }, {
            label: 'Cancel',
            onClick: handler,
            variation: BUTTON_VARIATION.SECONDARY
        }];
        const BottomSheet = BottomSheetForm(ContentComponent);
        const onClosePopup = () => true;
        const wrapper = mount(
            <Provider store={mockStore()}>
                <BottomSheet
                    formControls={controls}
                    id="defaultBottomSheet"
                    onClosePopup={onClosePopup}
                />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
