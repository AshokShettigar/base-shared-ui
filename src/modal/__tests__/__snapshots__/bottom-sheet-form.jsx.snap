// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal/bottom-sheet-form should render the HOC component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BottomSheetModal
    formControls={
      Array [
        Object {
          "label": "Save",
          "onClick": [Function],
          "variation": 1,
        },
        Object {
          "label": "Cancel",
          "onClick": [Function],
          "variation": 2,
        },
      ]
    }
    id="defaultBottomSheet"
    layoverClassName="cn-modal-layover"
    onClosePopup={[Function]}
    popupClassName="cn-modal cn-bottom-sheet"
  >
    <HigherOrderPopupComponent
      formControls={
        Array [
          Object {
            "label": "Save",
            "onClick": [Function],
            "variation": 1,
          },
          Object {
            "label": "Cancel",
            "onClick": [Function],
            "variation": 2,
          },
        ]
      }
      id="defaultBottomSheet"
      layoverClassName="cn-modal-layover"
      onClosePopup={[Function]}
      popupClassName="cn-modal cn-bottom-sheet"
      updatePopupProps={[Function]}
    />
  </BottomSheetModal>
</Provider>
`;

exports[`modal/bottom-sheet-form should render the component 1`] = `
<div
  className="cn-bottom-sheet-form"
>
  <div
    className="cn-bottom-sheet-form__body"
  >
    <div
      className="cn-modal__content"
    >
      <ContentComponent
        formControls={
          Array [
            Object {
              "label": "Save",
              "onClick": [Function],
              "variation": 1,
            },
            Object {
              "label": "Cancel",
              "onClick": [Function],
              "variation": 2,
            },
          ]
        }
        sidebar="Sidebar Content"
      />
    </div>
    <ModalSidebar>
      Sidebar Content
    </ModalSidebar>
  </div>
  <ModalControls
    controls={
      Array [
        Object {
          "label": "Save",
          "onClick": [Function],
          "variation": 1,
        },
        Object {
          "label": "Cancel",
          "onClick": [Function],
          "variation": 2,
        },
      ]
    }
  />
</div>
`;

exports[`modal/bottom-sheet-form should render the component with sub control 1`] = `
<div
  className="cn-bottom-sheet-form"
>
  <div
    className="cn-bottom-sheet-form__body"
  >
    <div
      className="cn-modal__content"
    >
      <ContentComponent
        formControls={
          Array [
            Object {
              "label": "Save",
              "onClick": [Function],
              "variation": 1,
            },
            Object {
              "label": "Cancel",
              "onClick": [Function],
              "variation": 2,
            },
          ]
        }
        subControl="Some control"
      />
    </div>
  </div>
  <div
    className="cn-modal-sub-controls"
  >
    Some control
  </div>
  <ModalControls
    controls={
      Array [
        Object {
          "label": "Save",
          "onClick": [Function],
          "variation": 1,
        },
        Object {
          "label": "Cancel",
          "onClick": [Function],
          "variation": 2,
        },
      ]
    }
  />
</div>
`;

exports[`modal/bottom-sheet-form should render the component without sidebar 1`] = `
<div
  className="cn-bottom-sheet-form"
>
  <div
    className="cn-bottom-sheet-form__body"
  >
    <div
      className="cn-modal__content"
    >
      <ContentComponent
        formControls={
          Array [
            Object {
              "label": "Save",
              "onClick": [Function],
              "variation": 1,
            },
            Object {
              "label": "Cancel",
              "onClick": [Function],
              "variation": 2,
            },
          ]
        }
      />
    </div>
  </div>
  <ModalControls
    controls={
      Array [
        Object {
          "label": "Save",
          "onClick": [Function],
          "variation": 1,
        },
        Object {
          "label": "Cancel",
          "onClick": [Function],
          "variation": 2,
        },
      ]
    }
  />
</div>
`;
